src/pool/srv_pool.c

```diff
@@ -1213,8 +1213,14 @@ pool_svc_step_up_cb(struct ds_rsvc *rsvc)
 	if (rc)
 		D_GOTO(out, rc);
 
-	uuid_generate(pool_hdl_uuid);
-	uuid_generate(cont_hdl_uuid);
+	if (!uuid_is_null(svc->ps_pool->sp_srv_cont_hdl)) {
+		uuid_copy(pool_hdl_uuid, svc->ps_pool->sp_srv_pool_hdl);
+		uuid_copy(cont_hdl_uuid, svc->ps_pool->sp_srv_cont_hdl);
+	} else {
+		uuid_generate(pool_hdl_uuid);
+		uuid_generate(cont_hdl_uuid);
+	}
+
```

Comment:
Does this _always_ work? E.g., a server who has not received the special pool or container handle in the first place suddenly becomes the PS leader? Generally, if we need to pass some info down future leaders, we write it to the RDB.

Reply:
oh, these uuid do not need to be constant, if they are changed due to new leader,  the request will retry until the new uuids are updated.  This is just optimization to avoid retry. so they do not need go into RDB.

Reply:
OK...

Reply:
Actually I don't quite see the value of propagating two two random handle uuids through a new IV.

For pool connection handle, I think we'd better go through the common pool connect interface (IV_POOL_CONN?), so that server credentials could be propagated to all target servers as well, I think it'll be useful for authentication.

For container open handle, I don't see why remote fetch needs a real open handle, the old of using special container handle for rebuild (or other remote fetch) looks better to me. (It looks to me that current patch and prior PR #3053 complicated things).

Reply:
oh, we need this for remote fetch in EC aggregation.  so we have to move this outside of rebuild.  Hmm, we do not have use cases for connect pool in server side yet.  But I do think IV propagate unique handle uuid is better than normal connect for server communications.  Could you please elaborate your concern?